src/serde_arrow_type.erl

```diff
@@ -48,27 +50,37 @@
 %% Any floating point number in Apache Arrow. Includes `Float 16', `Float 32'
 %% and `Float 64'.
 
+-type bin() :: bin.
+%% Any binary whose length in bits is a multiple of 8.
+
 -type erlang_type() ::
     boolean()
     | undefined
+    | nil
     | integer()
-    | float().
+    | float()
+    | binary().
 %% Any Erlang datatype which `serde_arrow' supports serializing from and
 %% deserializing into.
 
--spec bit_length(Type :: arrow_type()) -> Length :: pos_integer().
+-spec bit_length(Type :: arrow_type()) -> Length :: pos_integer() | undefined.
 %% doc Returns the size of the type in bits.
 bit_length({Type, Size}) when (Type =:= s) orelse (Type =:= u) orelse (Type =:= f) ->
     Size;
 bit_length(arrow_boolean) ->
-    1.
+    1;
+bit_length(bin) ->
+    %% This is a stub function.
+    undefined.
```

Comment:
Is this meant to be invoked with `bin` or it will never be invoked with `bin`? If the latter is the case, then it is better to raise (either explicit or by not defined the clause at all) rather than having `undefined` propagate through the system.

Reply:
So, this function will never be invoked with `bin`.  This function actually is not being used anywhere else (with any type) other than in byte_length. I only defined it in case we will need to work at the bit level for boolean buffers. The stub was written to shut up dialyzer.

However, there is a reason byte_length(bin) returns undefined. Later, when I use the length, I match for an undefined length and then find the actual length by calling byte_size on the binary there where I need the length.

Reply:
> However, there is a reason byte_length(bin) returns undefined. Later, when I use the length, I match for an undefined length and then find the actual length by calling byte_size on the binary there where I need the length.

Can you please link to the code? Because if that's the case... then it is not a stub anymore? Maybe it is worth updating the comment?

Reply:
So `bit_length(bin)` is a stub. It is not used anywhere, but only as of right now. I may use it once I provide proper support for boolean arrays. `byte_length(bin)` is not a stud:

https://github.com/Benjamin-Philip/serde_arrow/blob/fa34b479ae32dc3921eaf35b3d7050d149dc8232/src/serde_arrow_type.erl#L82-L83

It is used to decide how much to pad in case of a null value:

https://github.com/Benjamin-Philip/serde_arrow/blob/fa34b479ae32dc3921eaf35b3d7050d149dc8232/src/serde_arrow_buffer.erl#L51-L52

https://github.com/Benjamin-Philip/serde_arrow/blob/fa34b479ae32dc3921eaf35b3d7050d149dc8232/src/serde_arrow_buffer.erl#L81-L82

https://github.com/Benjamin-Philip/serde_arrow/blob/fa34b479ae32dc3921eaf35b3d7050d149dc8232/src/serde_arrow_buffer.erl#L86-L90

Reply:
So in my opinion we should raise here: `error(badarg)` or `error(bitlength)` or whatever. Prefer to make the unimplemented parts raise, rather than return dummy values, because a raise will be much easier to debug once you realize it is the missing bit.